src/seedu/addressbook/data/person/Address.java

```diff
@@ -33,7 +38,14 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
      * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+        String[] components = test.split(",\\s*");
+        if (components.length != 4) return false;
```

Comment:
Good check. but watch out for coding style! This is a dangerous way of styling your `if`